src/lib/trace/lttng/provider_circuit.h

```diff
@@ -0,0 +1,150 @@
+/* Copyright (c) 2020, The Tor Project, Inc. */
+/* See LICENSE for licensing information */
+
+/**
+ * \file provider_circuit.h
+ * \brief Header file that contains the LTTng probes for the tracepoint of the
+ *        circuit subsystem.
+ **/
+
+#undef TRACEPOINT_PROVIDER
+#define TRACEPOINT_PROVIDER tor_circuit
+
+#undef TRACEPOINT_INCLUDE
+#define TRACEPOINT_INCLUDE "./lib/trace/lttng/provider_circuit.h"
+
+#if !defined(_TRACEPOINT_TOR_CIRCUIT_H) || defined(TRACEPOINT_HEADER_MULTI_READ)
```

Comment:
We shouldn't define macros starting with _; those are reserved.

Also, why do we need this strange include guard?  If we can't do without it, we should have a comment.

Reply:
Also, what marks code tracepoint as lttng only?  It seems  to depend on lttng, but I don't see the "ifdef" for that.

Reply:
Yeah so those crazy ifdef are required by LTTng-UST: https://lttng.org/docs/v2.11/#doc-tpp-header

But we can easily control the `_TRACEPOIN...` so that I can change.